forte/pipeline.py

```diff
@@ -299,6 +321,33 @@ def _dump_to_config(self):
                     "configs": config.todict(),
                 }
             )
+
+        # Serialize current states of pipeline
+        configs.append(
```

Comment:
So this design mixes the component types and pipeline states at the same level. While this implementation works, it may not be easy to maintain, especially when we add something else in the future. My suggested configuration could be (basically separated the two into two dicts):
{
    “components”: {
        “type”: “xxx.NLTKSegmenter”
         ...
    }, # We store the components here
    “states”: {
        “_initialized”: True
        ...
    } # We store the states here
}

Reply:
As you also notice, this would affect some of our examples, so we can change them here:
https://github.com/asyml/forte-wrappers/blob/ac2d3715a4b86db44c025ef9155a63a1b979cf28/examples/gpt2/sample_multipack_pipeline_gpt.yml

Now it is good to start adding a version to the serialized pipeline, so that we know how to handle changes in the future